### DISCUSSION TITLE
Regex should work as expected in rewrites

### DISCUSSION THREAD

**Discussion by jeremylynch** [vercel/next.js#33165] [Help]
### What version of Next.js are you using?

11.1.3

### What version of Node.js are you using?

14.15.0

### What browser are you using?

Chrone

### What operating system are you using?

MacOS

### How are you deploying your application?

Vercel

### Describe the Bug

Regex should work as expected when capturing paths for rewrites or redirects. Eg. 

The following regex works fine `^[^.]*$` (from [stack overflow](https://stackoverflow.com/questions/4742357/how-to-match-only-strings-that-do-not-contain-a-dot-using-regular-expressions/4742369)) when finding any path containing dots (eg. Matches "sitemap" and NOT "sitemap.xml"). Test here: https://regexr.com/

```
REGEX = ^[^.]*$
sitemap.xml       NO MATCH ❌
sitemap              MATCH ✅
```

However, When I do the same in next.js, this does not work. See below: 

```js
async rewrites() {
    return {
      beforeFiles: [
        {
          source: '/:path(^[^.]*$)',
          destination: `/destination/:path`,
        },
      ]
    }
  },
```

### Expected Behavior

Should work as regexes work elsewhere

### To Reproduce

As per above

**Comment by mmojahed93** [comment]
I have the same problem :| 
No one can help us!

**Comment by leonllrmc** [comment]
Found a fix: using this expression instead `[^\\.]*` seams to work fine
